tests/MessageCountTests.cs

```diff
@@ -0,0 +1,69 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using FluentAssertions;
+using NUnit.Framework;
+using StreamChat.Models;
+
+namespace StreamChatTests
+{
+    [TestFixture]
+    public class MessageCountTests : TestBase
+    {
+        private UserRequest _user;
+        private ChannelWithConfig _channel;
+
+        [SetUp]
+        public async Task SetUp()
+        {
+            _user = await UpsertNewUserAsync();
+            _channel = await CreateChannelAsync(_user.Id, autoDelete: false);
+        }
+
+        [TearDown]
+        public async Task TearDown()
+        {
+            await TryDeleteChannelAsync(_channel.Cid);
+            await TryDeleteUsersAsync(_user.Id);
+        }
+
+        [Test]
+        public async Task TestMessageCountEnabledAsync()
+        {
+            await _messageClient.SendMessageAsync(_channel.Type, _channel.Id, _user.Id, "hello");
+            await WaitForAsync(async () =>
+            {
+                var chanState = await _channelClient.GetOrCreateAsync(_channel.Type, _channel.Id,
+                    new ChannelGetRequest { State = true });
+                return chanState.Channel.MessageCount == 1;
```

Comment:
The last issue is that this test depends implicitly on `count_messages` being enabled by default. If `TestMessageCountDisabledAsync` would run before `TestMessageCountEnabledAsync` then `TestMessageCountEnabledAsync` would fail because count messages would be disabled. 

Depending on default value is fine but the test should run in isolation, meaning that both tests should not use the same channel but rather use separate instances.